README.md

```diff
@@ -2,17 +2,129 @@
+  name: {{ .Namespace }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: another-service-account
+```
+
+Just add your templating data with the method `WithTemplating()`, and you're good to go! If you do like so, all occurrences of `.Namespace` in the given file will be rendered and then applied against the cluster.
+
+```go
+func yourCode() {
+  filename := "/your/fileWithGoTemplating.yaml"
+  yamlBytes := readFile(filename)
+   templateData := struct {
+     Namespace string
+   }{ Namespace: "your-namespace" }
+   
+  applier, _, err := apply.New(yourRestConfig)
+  err := applier.NewBuilder().
+    WithNamespace("your-namespace").
+    WithYamlResource(filename, doc).
+    WithTemplating(filename, templateData).
+    ExecuteApply()
+}
+```
+
+### Advanced: Owner Resources
+
+When working with your own CRDs inside a [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) garbage collection is a thing to be taken seriously. `k8s-apply-lib` provides a way of setting an owning resource. This way, if the owning resource is going to be deleted, the applied resources will be deleted as well. Please note, that setting an owner reference works only for namespace-scoped-to-namespace-scoped [ownership relations](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/). There can only be one owner per builder run.
+
+```go
+func yourCode() {
+  filename := "/your/fileWithGoTemplating.yaml"
+  yamlBytes := readFile(filename)
+  owner := &v1.ServiceAccount{
+      TypeMeta: metav1.TypeMeta{
+         APIVersion: "v1",
+         Kind:       "ServiceAccount",
+      },
+      ObjectMeta: metav1.ObjectMeta{
+         Name:      "le-service-account",
+      },
+   }
+   
+  applier, _, err := apply.New(yourRestConfig)
+  err := applier.NewBuilder().
+    WithNamespace("your-namespace").
+    WithOwner(owner).
+    WithYamlResource(filename, doc).
+    ExecuteApply()
+}
+```
+### Advanced: Resource Collection
+
+Sometimes a resource being applied to the Kubernetes API needs to be re-used somewhere else (f. i. a ServiceAccount must be mounted by name). `k8s-apply-lib` provides a way of matching and collecting resources while they stream through the `Applier`. `PredicatedResourceCollector` is an interface with two methods which you should implement to collect your resources:
+- `Predicate(doc YamlDocument) (bool, error)`
+  - should return true if the generic resource in the YAML document should be collected 
+- `Collect(doc YamlDocument)`
+  - takes care of the actual collection of your liking
+
+Please see the interface `PredicatedResourceCollector` in `Builder.go` for more information.
+
+```go
+func yourCode() {
+  filename := "/your/file.yaml"
+  yamlBytes := readFile(filename)
+   
+  applier, _, err := apply.New(yourRestConfig)
+  err := applier.NewBuilder().
+    WithNamespace("your-namespace").
+    WithYamlResource(filename, doc).
+    WithCollector(owner).
+    ExecuteApply()
```

Comment:
Even the simplest collection implementation is quite verbose because of the struct and two methods, not speaking of the error handling. That's why I refer to the code. I don't see anyone using the collector currently and would rather have a "Please document this bit" issue/PR then over-explaining everything.

What do you think about this?

Reply:
You actually provide an example for the collector implementation in the interface as documentation as mentioned above. I did not see this before. So it is good as it is.